tests/components/test_pilight.py

```diff
@@ -0,0 +1,296 @@
+"""The tests for the pilight component."""
+import logging
+import unittest
+from unittest.mock import patch
+import socket
+
+from homeassistant.bootstrap import setup_component
+from homeassistant.components import pilight
+
+from tests.common import get_test_home_assistant, assert_setup_component
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class PilightDaemonSim:
+    ''' Class to fake the interface of the pilight python package.
+
+    Is used in an asyncio loop, thus the mock cannot be accessed to
+    determine if methods where called?!
+    This is solved here in a hackish way by printing errors
+    that can be checked using logging.error mocks.
+    '''
```

Comment:
I do not like this approach, but did not find a better solution. The problem is that the
following does not work (most likely due to asyncio):

```python
  with unittest.mock.patch('pilight.pilight.Client') as mock_client:
    setup_component(self.hass, pilight.DOMAIN, {pilight.DOMAIN: {}})
    assert mock_client.called
```

The `pilight.pilight.Client` is correctly substituted with a `MagicMock()`, but the mock that is returned by the `unittest.mock.patch` function is not the one used in the hass instance.